src/npc.cpp

```diff
@@ -2428,6 +2429,20 @@ std::set<tripoint> npc::get_path_avoid() const
         // TODO: Cache this somewhere
         ret.insert( critter.pos() );
     }
+    if( rules.has_flag( ally_rule::avoid_doors ) ) {
+        for( const tripoint &p : g->m.points_in_radius( pos(), 30 ) ) {
```

Comment:
NPC will check a 30-tiles radius area every turn? Isn't it super-heavy strain on performance?

Reply:
Doesn't seem to be but my development rig is really beefy.  I suspect the fact that it trims the available options for pathfinding makes up for it.